### PR TITLE
Add organizational parameter value substitution.  [MTC-53]

### PR DESCRIPTION
**Pull request by woodt**

### CODE REVIEW

File: controls/models.py [hunk 1]
```diff
@@ -58,7 +57,7 @@ def catalog_control_as_dict(self):
         """Return the control content from the catalog"""
         # Get instance of the control catalog
         catalog = Catalog.GetInstance(catalog_key=self.sid_class)
-        catalog_control_dict = catalog.get_flattended_controls_all_as_dict()
```
Comment (gregelin) [inline controls/models.py:61]: Grateful for typo fix.

**Comment by gregelin** [comment]
@woodt I like the inclusion of the organizational defined parameters as part of instantiating the Catalog object. It is accurate to think of a catalog as having been fully defined by the including an org's parameter.

I think would probably be an easy way for us to instantiate and cache different Catalogs for different organizations, so we would get the performance while still having Catalogs for each org in an instance. (I wonder how we determine memory requirements for that).

File: controls/oscal.py [hunk 2]
```diff
@@ -54,15 +53,25 @@ class Catalog (object):
...
+
+        catalog_instance_key = '_cached_instance_' + catalog_key
 
-    def __init__(self, catalog_key='NIST_SP-800-53_rev4'):
+        if not hasattr(Catalog, catalog_instance_key):
+            new_catalog = Catalog(catalog_key=catalog_key, parameter_values=parameter_values)
+            setattr(Catalog, catalog_instance_key, new_catalog)
+        return getattr(Catalog, catalog_instance_key)
+
+    def __init__(self, catalog_key='NIST_SP-800-53_rev4', parameter_values=dict()):
```
Comment (bama4) [inline controls/oscal.py:74]: `NIST_SP-800-53_rev4` could be a constant instead of the hardcoded string

File: controls/tests.py [hunk 3]
```diff
@@ -200,13 +200,22 @@ class Oscal80053Tests(TestCase):
...
+        cg_flat = cg.get_flattened_controls_all_as_dict()
         control = cg_flat['au-2']
         self.assertEqual(control['id'].upper(), "AU-2")
         # self.assertEqual(control.class, "NIST.800.53")
         # TODO: ADD Class into object
         self.assertEqual(control['title'].upper(), "AUDIT EVENTS")
 
+    def test_catalog_controls_with_organizational_parameters(self):
+        parameter_values = { 'ac-1_prm_2': 'every 12 parsecs' }
+        cg = Catalog.GetInstance('NIST_SP-800-53_rev4')
```
Comment (bama4) [inline controls/tests.py:212]: `NIST_SP-800-53_rev4` could be a constant

File: controls/oscal.py (hunk 2 above)
Comment (gregelin) [reply controls/oscal.py:74]: Agreed @bama4

**Comment by gregelin** [comment]
@woodt This is looking very good. Glad to see the tests in place.

**Comment by gregelin** [comment]
@woodt We might want to address the database model for `Org_Catalog_Parameters` as a table with fields:

- organization
- catalog_key
- parameter_id
- label
- default
- default_type

If we could make the `organization` field arbitrary object type (as in attaching tags to different objects), then we could have one table for parameters whether related to organization, system, or portfolio.

File: controls/oscal.py (hunk 2 above)
Comment (woodt) [reply controls/oscal.py]: Made the catalog keys constants.

File: controls/tests.py (hunk 3 above)
Comment (woodt) [reply controls/tests.py]: Made catalog keys constants.

**Comment by woodt** [comment]
Cleaned up the API a bit by more closely tying org params to a catalog instance so the params don't need to be passed around
as much.  This now works because I changed the catalog instance cache to take into account the org parameters.  Added some basic persistence for organizational settings (we can add the additional fields Greg suggested when we add the features that require them.)  Removing the "draft" attribute from this PR.  I think it is ready for final review.

**Review by gregelin — APPROVED** [review]
Looks good. Code examined closely. Tests run. Small tweaks made.

**Pull request merged**